jetty-core/jetty-deploy/src/main/java/org/eclipse/jetty/deploy/DeploymentManager.java

```diff
@@ -53,178 +50,141 @@
  * <p>
  * <img alt="deployment manager roles graph" src="doc-files/DeploymentManager_Roles.png">
  * <ol>
- * <li>Tracking Apps and their LifeCycle Location</li>
- * <li>Managing AppProviders and the Apps that they provide.</li>
- * <li>Executing AppLifeCycle on App based on current and desired LifeCycle Location.</li>
+ * <li>Tracking ContextHandlers and their location in the Deployment graph.</li>
+ * <li>Moving ContextHandlers through the Deployment graph (eg: DEPLOYED, STARTED, UNDEPLOYED, etc.).</li>
  * </ol>
  * <p>
  * <img alt="deployment manager graph" src="doc-files/DeploymentManager.png">
  */
+// TODO: fix dumpable to show things like context-handler-collection, contexts being tracked, etc...
 @ManagedObject("Deployment Manager")
-public class DeploymentManager extends ContainerLifeCycle
+public class DeploymentManager extends ContainerLifeCycle implements ContextHandlerDeployer
```

Comment:
I think it would be good to have two implementations of `ContextHandlerDeployer`: a simple one, perhaps called `DirectDeploymentManager` that just does the minimal work to add a context.   Then we can have the `BindingDeploymentManager` that contains the graph and allows for binding on transitions etc.

Reply:
Nah, not worth it.
As even the `DirectDeploymentManager` is also participating in the graph and not doing direct deployment.